shared/oae/api/oae.api.tenancysearch.js

```diff
@@ -0,0 +1,129 @@
+    let query = location.attr('relative');
+
+    return (function () {
+        'use strict';
+
+        let searchTenancyModule =  {
+            init: function() {
+                this.cacheModule();
+                this.bindEvents();
+                this.render();
+            },
+            cacheModule: function() {
+                this.$module = $('#tenancyForm');
+                this.$input = this.$module.find('#tenancyInput');
+                this.$alert = this.$module.find('#tenancyAlert');
+            },
+            bindEvents: function() {
+                this.$input.on('select2:select', this.tenancySelect.bind(this));
+            },
+            render: function() {
+                this.$input.select2({
+                    dropdownParent: $('#signin-modal'),
+                    ajax: {
+                        url: '/api/search/tenants?limit=10',
+                        dataType: 'json',
+                        delay: 700,
+                        data: function (params) {
+                            return {
+                                q: params.term, // search term
+                                page: params.page
+                            };
+                        },
+                        error: function (data) {
+                            if (data.status === 0) {
+                                return;
+                            } else if ((data.status === 404 && data.statusText === 'Not Found') || (data.status === 400 && data.statusText === 'Bad Request')) {
+                                searchTenancyModule.$module.addClass('fail');
+                                searchTenancyModule.$input.select2('destroy');
+                            }
+                        },
+                        processResults: function (data, params) {
+                            // parse the results into the format expected by Select2
+                            // since we are using custom formatting functions we do not need to
+                            // alter the remote JSON data, except to indicate that infinite
+                            // scrolling can be used
+                            let tenancyResults = _.map(data.results, (eachResult) => {
+                                return {
+                                    host: eachResult.host,
+                                    displayName: eachResult.displayName,
+                                    id: eachResult.alias,
+                                    code: eachResult.countryCode
+                                };
+                            });
+
+                            tenancyResults = _.sortBy(tenancyResults, 'displayName');
+                            params.page = params.page || 1;
+
+                            return {
+                                results: tenancyResults,
+                                pagination: {
+                                    more: (params.page * 30) < data.totalCount
```

Comment:
`* 30`?

Reply:
oops. it's `10` now, to match the number of results being shown.